Src/BootCamp.Chapter1/ArrayOperations.cs

```diff
@@ -88,7 +204,35 @@ public static int[] InsertLast(int[] array, int number)
         /// <returns>A new array with element inserted at a given index. If an array is empty or null, returns new array with number in it.</returns>
         public static int[] InsertAt(int[] array, int number, int index)
         {
-            // ToDo: implement.
+            if (array != null && array.Length != 0)
+            {
+                var arrayLength = array.Length;
+                if (index < 0 || index > arrayLength)
+                {
+                    return array;
+                }
+
+                int[] newArray = new int[arrayLength + 1];
+                var addedIndex = 0;
+                // make a copy of array by value (not reference) and count/add if index added
+                for (var i = 0; i < arrayLength; i++)
+                {
+                    if (i == index)
+                    {
+                        newArray[i] = number;
+                        addedIndex = 1;
+                    }
+                    newArray[i + addedIndex] = array[i];
+                }
+                return newArray;
+            } else
+            {
+                // if array is null or 0, declare new array with the number at position 0
+                if (array != null || index < 0 || index > array.Length) return array;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['array' is null on at least one execution path.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5011611)